proposals/semi-auto-properties.md

```diff
@@ -1,14 +1,307 @@
 
 ## Motivation
-Standard auto-properties only allow for setting or getting the backing field directly, giving some control only by access modifying the accessor methods. Sometimes there is more need to have control over what happens when accessing an auto-property, without being confronted with all overhead of a standard property.
 
-Two common scenarios are that you want to apply a constraint on the setter, ensuring the validity of a value. The other being raising an event that informs about the property going to be changed/having been changed.
+Auto properties only allow for directly setting or getting the backing field, giving some control only by placing access modifiers on the accessors. Sometimes there is a need to have additional control over what happens in one or both accessors, but this confronts users with the overhead of declaring a backing field. The backing field name must then be kept in sync with the property, and the backing field is scoped to the entire class which can result in accidental bypassing of the accessors from within the class.
+
+There are two common scenarios in particular: applying a constraint on the setter to ensuring the validity of a value, and raising an event such as `INotifyPropertyChanged.PropertyChanged`.
+
+In these cases by now you always have to create an instance field and write the whole property yourself.  This not only adds a fair amount of code, but it also leaks the backing field into the rest of the type's scope, when it is often desirable to only have it be available to the bodies of the accessors.
+
+## Glossary
+
+- **Auto property**: Short for "automatically implemented property" ([§15.7.4](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/classes.md#1574-automatically-implemented-properties)). Accessors on an auto property have no body. The implementation and backing storage are both provided by the compiler. Auto properties have `{ get; }`, `{ get; set; }`, or `{ get; init; }`.
+
+- **Auto accessor**: Short for "automatically implemented accessor." This is an accessor that has no body. The implementation and backing storage are both provided by the compiler. `get;`, `set;` and `init;` are auto accessors.
+
+- **Full accessor**: This is an accessor that has a body. The implementation is not provided by the compiler, though the backing storage may still be (as in the example `set => field = value;`).
+
+## Detailed design
+
+For properties with an `init` accessor, everything that applies below to `set` would apply instead to the `init` accessor.
+
+There are two syntax changes:
+
+1. There is a new contextual keyword, `field`, which may be used within property accessor bodies to access a backing field for the property declaration.
+
+2. Properties may now mix and match auto accessors with full accessors. "Auto property" will continue to mean a property whose accessors have no bodies. None of the examples below will be considered auto properties.
```

Comment:
> Properties may now mix and match auto accessors with full accessors.

Was there an explicit language design decision to allow a mix within a property? I couldn't find a reference to this in the LDM notes.

Reply:
There was not an explicit decision that I'm aware of. This was part of the reason I wanted to have the LDM see this. @CyrusNajmabadi

Reply:
Let's add an explicit question to the "Open LDM questions" section below.

> 1. Can a property be declared with one auto-implemented accessor *and* one explicitly-implemented accessor.

I realize that case is included in the current question "Which of these scenarios should be allowed to compile?" but it would be good to be explicit about this specific case.

Reply:
i could have sworn this was already decided by ldm.

Reply:
Well, it was the entire feature name: semi-auto properties. This is a property that is half automatically implemented, half manually implemented. It was just not explicitly recorded. I added an open question so we can capture this, though I don't expect it to be seen as an update from the original intention.

Reply:
That's interesting. I think I assumed "semi-auto" in the feature name referred to the fact the backing field was automatically generated.